cmd/access/access_node_builder.go

```diff
@@ -178,57 +185,57 @@ func (builder *FlowAccessNodeBuilder) parseFlags() {
 // The factory function is later passed into the initMiddleware function to eventually instantiate the p2p.LibP2PNode instance
 func (builder *FlowAccessNodeBuilder) initLibP2PFactory(ctx context.Context,
 	nodeID flow.Identifier,
-	networkMetrics module.NetworkMetrics,
 	networkKey crypto.PrivateKey) (p2p.LibP2PFactoryFunc, error) {
 
-	// setup the Ping provider to return the software version and the sealed block height
-	pingProvider := p2p.PingInfoProviderImpl{
-		SoftwareVersionFun: func() string {
-			return build.Semver()
-		},
-		SealedBlockHeightFun: func() (uint64, error) {
-			head, err := builder.State.Sealed().Head()
+	return func() (*p2p.Node, error) {
+		host, err := p2p.LibP2PHost(ctx, builder.unstakedNetworkBindAddr, networkKey, p2p.WithLibP2PPing(false))
+		if err != nil {
+			return nil, err
+		}
+
+		var discovery *discovery.RoutingDiscovery
+		if builder.IsStaked() {
+			discovery, err = p2p.NewDHTServer(ctx, host)
 			if err != nil {
-				return 0, err
+				return nil, err
 			}
-			return head.Height, nil
-		},
-	}
+		} else {
+			discovery, err = p2p.NewDHTClient(ctx, host)
+			if err != nil {
+				return nil, err
+			}
+		}
 
-	libP2PNodeFactory, err := p2p.DefaultLibP2PNodeFactory(ctx,
-		builder.Logger,
-		nodeID,
-		builder.unstakedNetworkBindAddr,
-		networkKey,
-		builder.RootBlock.ID().String(),
-		p2p.DefaultMaxPubSubMsgSize,
-		networkMetrics,
-		pingProvider)
-	if err != nil {
-		return nil, fmt.Errorf("could not generate libp2p node factory: %w", err)
-	}
+		psOption := libp2ppubsub.WithDiscovery(discovery)
 
-	return libP2PNodeFactory, nil
+		pubsub, err := p2p.DefaultPubSub(ctx, host, psOption)
+		if err != nil {
+			return nil, err
+		}
+
+		builder.UnstakedLibP2PNode, err = p2p.NewLibP2PNode(nodeID, builder.RootBlock.ID().String(), builder.Logger, host, pubsub)
+		if err != nil {
+			return nil, err
+		}
+		return builder.UnstakedLibP2PNode, nil
+	}, nil
 }
 
 // initMiddleware creates the network.Middleware implementation with the libp2p factory function, metrics, peer update
 // interval, and validators. The network.Middleware is then passed into the initNetwork function.
 func (builder *FlowAccessNodeBuilder) initMiddleware(nodeID flow.Identifier,
 	networkMetrics module.NetworkMetrics,
 	factoryFunc p2p.LibP2PFactoryFunc,
-	peerUpdateInterval time.Duration,
-	connectionGating bool,
-	managerPeerConnections bool,
 	validators ...network.MessageValidator) *p2p.Middleware {
 	builder.unstakedMiddleware = p2p.NewMiddleware(builder.Logger,
 		factoryFunc,
 		nodeID,
 		networkMetrics,
 		builder.RootBlock.ID().String(),
-		peerUpdateInterval,
+		time.Hour, // TODO: this is pretty meaningless since there is no peermaanger in play.
```

Comment:
will be addressed via [issue](https://github.com/dapperlabs/flow-go/issues/5713)